core/vibes/soul/sections/product-detail/index.tsx

```diff
@@ -8,154 +8,147 @@ import { ProductGallery } from '@/vibes/soul/sections/product-detail/product-gal
 import { ProductDetailForm, ProductDetailFormAction } from './product-detail-form';
 import { Field } from './schema';
 
-interface ProductDetailProduct {
-  id: string;
-  title: string;
-  href: string;
-  images: Streamable<Array<{ src: string; alt: string }>>;
-  price?: Streamable<Price | null>;
-  subtitle?: string;
-  badge?: string;
+interface Props<F extends Field> {
+  productId: string;
+  breadcrumbs?: Streamable<Breadcrumb[]>;
+  title: Streamable<string>;
+  subtitle?: Streamable<string | null>;
   rating?: Streamable<number | null>;
+  price?: Streamable<Price | null>;
+  images: Streamable<Array<{ src: string; alt: string }>>;
   summary?: Streamable<string>;
+  fields: Streamable<F[]>;
   description?: Streamable<string | React.ReactNode | null>;
   accordions?: Streamable<
     Array<{
       title: string;
       content: React.ReactNode;
     }>
   >;
-}
-
-interface Props<F extends Field> {
-  breadcrumbs?: Streamable<Breadcrumb[]>;
-  product: Streamable<ProductDetailProduct | null>;
-  action: ProductDetailFormAction<F>;
-  fields: Streamable<F[]>;
   quantityLabel?: string;
   incrementLabel?: string;
   decrementLabel?: string;
   ctaLabel?: Streamable<string | null>;
   ctaDisabled?: Streamable<boolean | null>;
+  action: ProductDetailFormAction<F>;
   prefetch?: boolean;
   thumbnailLabel?: string;
 }
 
 export function ProductDetail<F extends Field>({
-  product: streamableProduct,
-  action,
-  fields: streamableFields,
+  productId,
   breadcrumbs,
+  title: streamableTitle,
+  subtitle: streamableSubtitle,
+  rating: streamableRating,
+  price: streamablePrice,
+  images: streamableImages,
+  fields: streamableFields,
+  summary: streamableSummary,
+  description: streamableDescription,
+  accordions: streamableAccordions,
   quantityLabel,
   incrementLabel,
   decrementLabel,
   ctaLabel: streamableCtaLabel,
   ctaDisabled: streamableCtaDisabled,
+  action,
   prefetch,
   thumbnailLabel,
 }: Props<F>) {
   return (
     <section className="@container">
       <div className="mx-auto w-full max-w-screen-2xl px-4 py-10 @xl:px-6 @xl:py-14 @4xl:px-8 @4xl:py-20">
         {breadcrumbs && <Breadcrumbs breadcrumbs={breadcrumbs} className="mb-6" />}
-
```

Comment:
Best reviewed with the "Hide whitespace" toggle on:

<img width="236" alt="hide whitespace" src="https://github.com/user-attachments/assets/726cb134-c9f4-4f8c-8964-bda6076ea438" />